src/main/java/org/jenkinsci/plugins/workflow/cps/InterceptingExecutorService.java

```diff
@@ -0,0 +1,84 @@
+package org.jenkinsci.plugins.workflow.cps;
+
+import com.google.common.util.concurrent.ForwardingExecutorService;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * {@link ExecutorService} that wraps all the tasks that run inside.
+ *
+ * @author Kohsuke Kawaguchi
+ * @since 1.557
+ */
+public abstract class InterceptingExecutorService extends ForwardingExecutorService {
```

Comment:
If possible, `@Restricted(NoExternalUse.class)`, plenty of plugins depend on this one.